pkg/cleaner/cleaning.go

```diff
@@ -105,18 +108,31 @@ func (c *FileContentObfuscator) ObfuscateFile(inputFile string, outputFile strin
 	// we need to assess whether the file exists already to ensure we don't overwrite existing obfuscated data.
 	// that can happen while obfuscating file names and their paths.
 	// Additionally, the stat check is required because os.O_CREATE will implicitly os.O_TRUNC if a file already exist
+	// Also defining a lock to avoid collisions while creation of files in a multi-threaded environment
+	var mutexLock sync.Mutex
+	mutexLock.Lock()
```

Comment:
always accompany your lock with a defer unlock. In your code when we return through an error it would keep the lock hold forever.

you can also scope locks through defers when you wrap them in functions:
https://kylewbanks.com/blog/when-to-use-defer-in-go

which I would suggest in this case, you are passing a writePath and you return a file that does not collide.